### PR TITLE
Network Monitoring, Infra Health, Database Query Profiling Tutorials

### PR DESCRIPTION
**Pull request by htroisi**

### CODE REVIEW

**Review by aimichelle — DISMISSED** [review]

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+---
+title: "Network Monitoring"
+metaTitle: "Using Pixie | Use Cases | Network Monitoring"
+metaDescription: "Network monitoring using Pixie."
+order: 1
+---
+
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: This needs an engaging image at the top IMO

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+This tutorial will demonstrate how to use Pixie to:
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: Maybe add a preamble sentence here? Like Diagnosing issues at the network layer of your stack can be challenging without proper visibility into the traffic. 

"After deploying Pixie, you can easily:"

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+- [Visualize the flow of network traffic within your cluster.](#visualize-network-traffic-flowing-within-the-cluster)
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: Linking these seems like overkill, especially with the repeated use of "Visualize". it gives the impression that there is going to be a dense amount of content for a 101 tutorial.

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+- [Visualize the flow of DNS requests within your cluster.](#visualize-dns-requests-made-within-the-cluster)
+- [Visualize TCP drops and TCP retransmits across your cluster.](#visualize-tcp-drops-and-tcp-retransmissions-across-the-cluster)
+
+**Prerequisites**
+
+1. You will need a Kubernetes cluster with Pixie installed. If you do not have a cluster, you can create a minikube cluster and install Pixie using our [install guides](/installing-pixie/quick-start/).
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: heads up, this will have to be updated when my install guide changes land. you may want to link /installing-pixie and call it "installation steps"

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+- Quickly determine which services are communicating.
+- Identify any misconfigured services.
+- Identify load balancing issues.
+
+Let’s use the `px/net_flow_graph` script to see a graph of all of the network traffic passing through the cluster:
+
+1. Open the [Live UI](http://work.withpixie.ai/) and select `px/net_flow_graph` from the `script` drop-down menu at the top.
+
+> When the script opens, you’ll get an error indicating a value is missing for the `namespace` required argument. The UI denotes required script arguments with an asterisk after the argument name.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: maybe make this an info?

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+::: div image-l
+<svg title='' src='use-case-tutorials/missing_required_arg.png'/>
+:::
+
+2. Enter `pl` for the required `namespace` argument: select the drop down arrow next to the `namespace` argument, type `pl`, and hit enter.  `pl` is the namespace that Pixie deploys to.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: break into multiple lines?

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+<PoiTooltip top={60} left={87}>
+<strong>Enable Hierarchy View</strong>
+{' '}
+for a different view of the graph.
+</PoiTooltip>
+
+<svg title='' src='use-case-tutorials/net_flow_graph.png'/>
+:::
+
+> This script shows a mapping of all the outgoing connections for the pods in the specified namespace.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: Move this to above the image

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+> Grey hexagonal icons represent pods making network requests. Blue circles represent remote endpoint. If a request’s remote endpoint is within the cluster, then the IP address is resolved to a pod/service name.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: This seems like it could possibly be a tooltip

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+pan, zoom, or rearrange individual nodes.
+</PoiTooltip>
+
+<PoiTooltip top={37} left={62}>
+<strong>Hover over an Edge</strong>
+{' '}
+for network transmission stats. Thicker lines indicate more traffic.
+</PoiTooltip>
+
+<PoiTooltip top={60} left={87}>
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: Add "Hover over the flashing icons to see tips about this view" soemwhere

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+<svg title='' src='use-case-tutorials/net_flow_graph.png'/>
+:::
+
+> This script shows a mapping of all the outgoing connections for the pods in the specified namespace.
+
+> Grey hexagonal icons represent pods making network requests. Blue circles represent remote endpoint. If a request’s remote endpoint is within the cluster, then the IP address is resolved to a pod/service name.
+
+> Let's filter the graph to only show communication to the `pl-nats` pod.
+
+3. Select drop down arrow next to the `to_entity_filter` argument, type `pl-nats`, and press enter to re-run the script.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: `Enter`

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+> The graph should update to only show network traffic sent to the `pl-nats` pod. Pixie uses NATS as our messaging system.
+
+4. Clear the `to_entity_filter` value by selecting the drop-down arrow and pressing enter.
+
+5. Scroll down to the table below the graph. This table contains the same data that is used to construct the graph above.
+
+## Live DNS request map
+
+A global view of DNS requests made within a cluster can be used to:
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: I would rephrase so it doesn't sound canned w/the above similar language.

Another capability Pixie provides is the ability to inspect and analyze DNS traffic. Here are somethings Pixie can help you do:

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+> This script shows all of the DNS requests made in the cluster, with latency and throughput stats.
+
+> This cluster doesn't have anything deployed except Pixie, so all we see are the kube-dns pods communicating with the metadata server.
+
+2. Click on the `LATENCY_AVG` column title to sort the table data by average latency.
+
+## Live TCP drops, retransmits map
+
+TCP drops and retransmits can indicate network connectivity issues that may affect application performance.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: This is a good pre-amble

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+<svg title='' src='use-case-tutorials/dns_flow_graph.png'/>
+:::
+
+> This script shows all of the DNS requests made in the cluster, with latency and throughput stats.
+
+> This cluster doesn't have anything deployed except Pixie, so all we see are the kube-dns pods communicating with the metadata server.
+
+2. Click on the `LATENCY_AVG` column title to sort the table data by average latency.
+
+## Live TCP drops, retransmits map
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: might say TCP retransmits for extra clarity even to repeat it

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 3 above)
Comment (nserrino) [reply content/en/05-use-case-tutorials/network-monitoring.md]: especially since it already gets a linked outline on the right

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 13 above)
Comment (nserrino) [reply content/en/05-use-case-tutorials/network-monitoring.md]: not sure all of these titles need to start w/ Live, not sure it's useful

File: content/en/05-use-case-tutorials/network-monitoring.md [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+TCP drops and retransmits can indicate network connectivity issues that may affect application performance.
+
+Let's use the `px/tcp_drops`script to see a global view of TCP drops across the cluster.
+
+1. Select `px/tcp_drops` from the script drop-down menu.
+
+> This script is different from other PxL scripts. The two previous scripts simply queried the Pixie platform for data that is being continuously collected (DNS events, HTTP events, etc). This script deploys a new short-lived data source, then queries the new data source after some time has passed.
```
Comment (nserrino) [inline content/en/05-use-case-tutorials/network-monitoring.md]: i would make this an info

Comment (nserrino) [reply content/en/05-use-case-tutorials/network-monitoring.md]: This script is a PxL mutation script. While the previous scripts query the Pixie platform for data, this script extends Pixie to collect new data.

**Review by nserrino — CHANGES_REQUESTED** [review]
Review of the Network Monitoring tutorial

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 5 above)
Comment (htroisi) [reply content/en/05-use-case-tutorials/network-monitoring.md]: I tried that and I didn't look good.

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 2 above)
Comment (htroisi) [reply content/en/05-use-case-tutorials/network-monitoring.md]: Is this better? 

"Application health and network performance are inextricably linked. With Pixie, you can easily monitor your network alongside your applications and infrastructure.

This tutorial will demonstrate how to use Pixie to:"

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 3 above)
Comment (htroisi) [reply content/en/05-use-case-tutorials/network-monitoring.md]: Hyperlinks are removed. I replaced visualize with graph for slightly less fancy wording.

File: content/en/05-use-case-tutorials/network-monitoring.md (hunk 2 above)
Comment (nserrino) [reply content/en/05-use-case-tutorials/network-monitoring.md]: Network performance can have a big impact on the health of your services. However, problems in network performance can often be hard to detect when debugging. With Pixie, you can easily monitor your network alongside your applications and infrastructure.

**Review by nserrino — APPROVED** [review]
lgtm

**Review by aimichelle — APPROVED** [review]
